### ISSUE TITLE
TypeError When Changing to Dark Mode

### ISSUE THREAD

**Issue by danyeaw** [gaphor/gaphor#2860]
### Describe the bug
I changed my system manually to override to light mode, and then back to dark mode and I got a TypeError.

### To Reproduce
1. Override to light mode
2. Open another model
3. Override to dark mode
4. See Error

If possible, add the error report from `≡` (top-right menu) → `Tools` → `Error Reports`.

```
Gaphor version:         2.21.0
Operating System:       Linux (6.6.1-1-default)
Display:                GdkWaylandDisplay
Python version:         3.12.0
GTK version:            4.12.3
Adwaita version:        1.4.0
GtkSourceView version:  5.10.0
Cairo version:          1.18.0
Pango version:          1.51.0
PyGObject version:      3.46.0
Pycairo version:        1.25.1
pygit2/libgit2 version: 1.13.2  / 1.7.1

|Traceback (most recent call last):
|  File "/home/dan/Projects/gaphor/gaphor/ui/diagrampage.py", line 200, in _on_notify_dark
|    self.update_drawing_style()
|    └ <gaphor.ui.diagrampage.DiagramPage object at 0x7fd6a1b58050>
|  File "/home/dan/Projects/gaphor/gaphor/ui/diagrampage.py", line 238, in update_drawing_style
|    style = self.diagram.style(StyledDiagram(self.diagram, dark_mode=dark_mode))
|            │                  │             │                       └ True
|            │                  │             └ <gaphor.ui.diagrampage.DiagramPage object at 0x7fd6a1b58050>
|            │                  └ <class 'gaphor.core.modeling.diagram.StyledDiagram'>
|            └ <gaphor.ui.diagrampage.DiagramPage object at 0x7fd6a1b58050>
|  File "/home/dan/Projects/gaphor/gaphor/core/modeling/diagram.py", line 280, in style
|    style_sheet = self.styleSheet
|                  └ <gaphor.core.modeling.diagram.Diagram element DCE:AEC6FCA4-4586-11DA-A04F-00123FE76EBE>
|  File "/home/dan/Projects/gaphor/gaphor/core/modeling/diagram.py", line 277, in styleSheet
|    return next(self.model.select(StyleSheet), None)
|                │                 └ <class 'gaphor.core.modeling.stylesheet.StyleSheet'>
|                └ <gaphor.core.modeling.diagram.Diagram element DCE:AEC6FCA4-4586-11DA-A04F-00123FE76EBE>
|  File "/home/dan/Projects/gaphor/gaphor/core/modeling/element.py", line 110, in model
|    raise TypeError(
|TypeError: Can't retrieve the model since it's not set on construction
```

### Expected behavior
No error

### OS
- [X] Linux
- [ ] macOS
- [ ] Windows

### Version
Version of Gaphor: main

**Comment by amolenaar** [comment]
Did you close the first model?

**Comment by amolenaar** [comment]
Can't reproduce it on macOS at least. I discovered that the preference is always set to "System" when I open the Preferences dialog (on macOS).

**Comment by amolenaar** [comment]
The problem is that `DiagramPage.close()` is not called when a window is closed.

(I just had to install the schemas for the preferences dialog to work on Linux)

**Comment by amolenaar** [comment]
It looks like the schema is not properly loaded for our distributions. At least for Flatpak and macOS.

**Comment by danyeaw** [comment]
I am noticing that the Preferences are resetting to System as well

**Comment by danyeaw** [comment]
This issue is now fixed by #2875. I'll open a new issue for preferences not saving.
